packages/babel-preset/src/transformPlugin.ts

```diff
@@ -12,45 +18,60 @@ import { evaluatePaths } from './utils/evaluatePaths';
 import { BabelPluginOptions } from './types';
 import { validateOptions } from './validateOptions';
 
+type FunctionKinds = 'makeStyles' | 'makeResetStyles';
+
 type BabelPluginState = PluginPass & {
   importDeclarationPaths?: NodePath<t.ImportDeclaration>[];
   requireDeclarationPath?: NodePath<t.VariableDeclarator>;
 
-  definitionPaths?: NodePath<t.ObjectExpression>[];
+  definitionPaths?: { functionKind: FunctionKinds; path: NodePath<t.Expression | t.SpreadElement> }[];
   calleePaths?: NodePath<t.Identifier>[];
 };
 
-function getDefinitionPathFromMakeStylesCallExpression(
+function getDefinitionPathFromCallExpression(
+  functionKind: FunctionKinds,
   callExpression: NodePath<t.CallExpression>,
-): NodePath<t.ObjectExpression> {
+): NodePath<t.Expression | t.SpreadElement> {
   const argumentPaths = callExpression.get('arguments') as NodePath<t.Node>[];
   const hasValidArguments = Array.isArray(argumentPaths) && argumentPaths.length === 1;
 
   if (!hasValidArguments) {
-    throw new Error('makeStyles() function accepts only a single param');
+    throw callExpression.buildCodeFrameError(`${functionKind}() function accepts only a single param`);
   }
 
   const definitionsPath = argumentPaths[0];
 
-  if (!definitionsPath.isObjectExpression()) {
-    throw definitionsPath.buildCodeFrameError('makeStyles() function accepts only an object as a param');
+  if (functionKind === 'makeStyles') {
+    if (!definitionsPath.isObjectExpression()) {
+      throw definitionsPath.buildCodeFrameError('makeStyles() function accepts only an object as a param');
```

Comment:
both makeStyles and makeResetStyles accept an object only as param right ?

Reply:
Correct, will check it.

Reply:
Restored to the previous condition, don't remember why it looks like that